tests/test_modeling_common.py

```diff
@@ -1539,6 +1552,44 @@ def check_same_values(layer_1, layer_2):
             # self.assertTrue(model.transformer.wte.weight.shape, model.lm_head.weight.shape)
             # self.assertTrue(check_same_values(model.transformer.wte, model.lm_head))
 
+    def test_tied_model_weights_key_ignore(self):
+        config, inputs_dict = self.model_tester.prepare_config_and_inputs_for_common()
+        for model_class in self.all_model_classes:
+            model_tied = model_class(config)
+            with tempfile.TemporaryDirectory() as d:
+                model_tied.save_pretrained(d)
+
+                # We are nuking ALL weights on file, so every parameter should
+                # yell on load. We're going to detect if we yell too much, or too little.
+                with open(os.path.join(d, "pytorch_model.bin"), "wb") as f:
+                    torch.save({}, f)
+                model_reloaded, infos = model_class.from_pretrained(d, output_loading_info=True)
+
+                prefix = f"{model_reloaded.base_model_prefix}."
+                params = dict(model_reloaded.named_parameters())
+                params.update(dict(model_reloaded.named_buffers()))
```

Comment:
You mentioned on DM 

> They are listed as two different things during from_pretrained but only appear once when doing model.named_parameters()

I haven't 100% understand it (despite got some rough feeling about it). Could you explain

- why not using `model_reloaded.state_dict()` -- (from your word, I guess we will get 2 things, and it will match the missing keys, which will pass the test when the situation should fail)
- Give an example, say what happens to bert or gpt2

In any case, a comment (inside the code) should be added here regarding the usage of `named_parameters` and `named_parameters` 🙏 .

Reply:
As long as using named_parameters/named_buffer works, I wouldn't request to this fancier check :-)